eth-connector/src/lib.rs

```diff
@@ -213,6 +214,188 @@ impl FungibleTokenCore for EthConnectorContract {
     }
 }
 
+/// Fungible Token Trait implementation for compatibility with Engine NEP-141 methods.
+/// It's because we should ve known correct `sender_id`. In reference
+/// implementation it's `predecessor_account_id`. To resolve it
+/// we just set `sender_id` explicitly as function parameter.
+/// Also we check access right to manage access rights.
+#[near_bindgen]
+impl EngineFungibleToken for EthConnectorContract {
+    #[payable]
+    fn engine_ft_transfer(
+        &mut self,
+        sender_id: AccountId,
+        receiver_id: AccountId,
+        amount: U128,
+        memo: Option<String>,
+    ) {
+        self.assert_access_right().sdk_unwrap();
+        self.register_if_not_exists(&receiver_id);
+        assert_one_yocto();
+        let amount: Balance = amount.into();
+        self.ft
+            .internal_transfer(&sender_id, &receiver_id, amount, memo);
+    }
+
+    #[payable]
+    fn engine_ft_transfer_call(
+        &mut self,
+        sender_id: AccountId,
+        receiver_id: AccountId,
+        amount: U128,
+        memo: Option<String>,
+        msg: String,
+    ) -> PromiseOrValue<U128> {
+        self.assert_access_right().sdk_unwrap();
+        self.register_if_not_exists(&receiver_id);
+
+        let amount: Balance = amount.into();
+        crate::log!(
+            "Transfer call from {} to {} amount {}",
+            sender_id,
+            receiver_id,
+            amount,
+        );
+
+        // Verify message data before `ft_on_transfer` call to avoid verification panics
+        // It's allowed empty message if `receiver_id =! current_account_id`
+        if sender_id == receiver_id {
+            let message_data = FtTransferMessageData::parse_on_transfer_message(&msg).sdk_unwrap();
+            // Check is transfer amount > fee
+            if message_data.fee.as_u128() >= amount {
+                panic_err("insufficient balance for fee");
+            }
+        }
+
+        // Special case for Aurora transfer itself - we shouldn't transfer
```

Comment:
Unclear comment, IMO

Reply:
In that case it doesn't matter. It's refactored in #12